ops/testing.py

```diff
@@ -1275,9 +1310,15 @@ def __init__(self, backend: _TestingModelBackend):
         # Has a service been started/stopped?
         self._service_status = {}
         self._fs = _MockFilesystem()
+        self._backend = backend
+
+    def _check_connection(self):
+        if not self._backend._can_connect(self):
+            raise pebble.ConnectionError('cannot connect to pebble')
```

Comment:
Alertmanager tests fail because of this `raise` here.
I wonder why this wrapper is needed instead of using `_can_connect` directly.

Reply:
Looks like there is some `@patch`ing causing the issue.